File: pkg/server/pokete/fight/poke.go
package fight

type Nature struct {
	Nature string `json:"nature"`
	Grade  uint   `json:"grade"`
}

type Poke struct {
	Name    string   `json:"name"`
	Xp      uint32   `json:"xp"`
	Hp      uint32   `json:"hp"`
	Ap      [4]uint  `json:"ap"`
	Effects []string `json:"effects"`
	Attacks []string `json:"attacks"`
	Shiny   bool     `json:"shiny"`
	Nature  Nature   `json:"nature"`
}

File: pokete_classes/multiplayer/msg/poke.py
from typing import TypedDict


class StatsDict(TypedDict):
    ownership_date: str | None
    evolved_date: str | None
    total_battles: int
    lost_battles: int
    win_battles: int
    earned_xp: int
    caught_with: str
    run_away: int


class NatureDict(TypedDict):
    nature: str
    grade: int


class PokeDict(TypedDict):
    name: str
    xp: int
    hp: int
    ap: list[int]
    effects: list[str]
    attacks: list[str]
    shiny: bool
    nature: NatureDict
    stats: StatsDict
